data_validation/clients.py

```diff
@@ -155,28 +172,29 @@ def get_ibis_query(client, query):
     return iq
 
 
-def get_ibis_table_schema(client, schema_name, table_name):
+def get_ibis_table_schema(client, schema_name: str, table_name: str) -> "sch.Schema":
     """Return Ibis Table Schema for Supplied Client.
 
     client (IbisClient): Client to use for table
     schema_name (str): Schema name of table object, may not need this since Backend uses database
     table_name (str): Table name of table object
     database_name (str): Database name (generally default is used)
     """
-    if client.name in [
-        "mysql",
-        "oracle",
-        "postgres",
-        "db2",
-        "mssql",
-        "redshift",
-        "snowflake",
-    ]:
+    if client.name in IBIS_ALCHEMY_BACKENDS:
         return client.table(table_name, schema=schema_name).schema()
     else:
         return client.get_schema(table_name, schema_name)
 
 
+def get_ibis_query_schema(client, query_str) -> "sch.Schema":
+    if client.name in IBIS_ALCHEMY_BACKENDS:
+        ibis_query = get_ibis_query(client, query_str)
+        return ibis_query.schema()
+    else:
+        # NJ: I'm not happy about calling a private method but don't see how I can avoid it.
+        return client._get_schema_using_query(query_str)
```

Comment:
Ibis does not expose a public method like it does for get_schema(). I didn't feel it was right for me to add a public method to Ibis objects, I know we already patch the private method but adding new methods seemed a step too far. Happy to be influenced in another direction though.

Reply:
Well, since you added a comment about this current implementation I'd be okay to move forward as it is so we can unblock the customer. It seems like something we can analyze further later on.